lib/block-supports/layout.php

```diff
@@ -173,6 +194,11 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 	$block_classname    = wp_get_block_default_classname( $block['blockName'] );
 	$container_class    = wp_unique_id( 'wp-container-' );
 	$layout_classname   = '';
+	$use_global_padding = isset( $used_layout['inherit'] ) && $used_layout['inherit'] || isset( $used_layout['contentSize'] ) && $used_layout['contentSize'];
```

Comment:
Same feedback as the JS implementation. If I switch off `useRootPaddingAwareAlignments` in `theme.json` the classname still gets added:

<img width="638" alt="image" src="https://user-images.githubusercontent.com/14988353/178879921-3294e498-4238-4902-9e44-e69067d4796d.png">

Would adding another `gutenberg_get_global_settings` line above to check for the setting work?

Reply:
Done!